src/coreclr/vm/jitinterface.cpp

```diff
@@ -11915,12 +11960,27 @@ bool CEEInfo::getReadonlyStaticFieldValue(CORINFO_FIELD_HANDLE fieldHnd, uint8_t
             if (fieldObj != NULL)
             {
                 Object* obj = OBJECTREFToObject(fieldObj);
+                size_t handle;
                 if (GCHeapUtilities::GetGCHeap()->IsInFrozenSegment(obj))
                 {
-                    intptr_t ptr = (intptr_t)obj;
-                    memcpy(buffer, &ptr, sizeof(intptr_t));
+                    handle = (size_t)obj;
+                    memcpy(buffer, &handle, sizeof(size_t));
                     result = true;
                 }
+                else if (!ignoreMovableObjects)
+                {
+                    PTR_MethodTable objMT = obj->GetMethodTable();
+                    // We don't need to limit it to these types but JIT doesn't need other types of
+                    // objects at the moment so it's for better throughput.
+                    if (objMT->IsString() || objMT->IsArray())
+                    {
```

Comment:
```suggestion
```
I think the condition should not be needed now that we have `ignoreMovableObjects` flag